SingularityUI/app/actions/api/base.es6

```diff
@@ -46,9 +46,20 @@ export function buildApiAction(actionName, opts={}) {
 
       let options = optsFunc(...args);
       return fetch(config.apiRoot + options.url, _.extend({credentials: 'include'}, _.omit(options, 'url')))
-        .then(response => response.json())
-        .then(json => {
-          dispatch(success(json));
+        .then(response => {
+          if (response.status >= 200 && response.status < 300) {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(json => dispatch(success(json)));
+            } else {
+              return response.text().then(body => dispatch(success({response: body})));
+            }
+          } else {
+            if (response.headers.get('Content-Type') === 'application/json') {
+              return response.json().then(body => dispatch(error(body)));
+            } else {
+              return response.text().then(body => dispatch(error({message: body})));
+            }
+          }
```

Comment:
this block of code smells weird, but I couldn't put my finger on it initially... You can simplify the code here by chaining these together instead... i.e. `.then(response => response.json())` on line 49 would update to get the `Content-Type` check, and then a separate `.then()` to throw an exception that is caught lower down -- does that make sense?